applications/dashboard/src/scripts/compatibilityStyles/clickableItemHelpers.ts

```diff
@@ -0,0 +1,178 @@
+/**
+ * Compatibility styles, using the color variables.
+ *
+ * @copyright 2009-2019 Vanilla Forums Inc.
+ * @license GPL-2.0-only
+ */
+import {
+    colorOut,
+    ColorValues,
+    ILinkColorOverwritesWithOptions,
+    linkStyleFallbacks,
+} from "@library/styles/styleHelpers";
+import { cssOut, nestedWorkaround, trimTrailingCommas } from "@dashboard/compatibilityStyles/index";
+import { globalVariables } from "@library/styles/globalStyleVars";
+import { NestedCSSProperties } from "typestyle/lib/types";
+import merge from "lodash/merge";
+import { emptyObject } from "expect/build/utils";
+import { NestedCSSSelectors } from "typestyle/src/types";
+
+export const EMPTY_STATE_COLORS = {
+    default: undefined as undefined | ColorValues,
+    hover: undefined as undefined | ColorValues,
+    focus: undefined as undefined | ColorValues,
+    clickFocus: undefined as undefined | ColorValues,
+    keyboardFocus: undefined as undefined | ColorValues,
+    active: undefined as undefined | ColorValues,
+    visited: undefined as undefined | ColorValues,
+    allStates: undefined as undefined | ColorValues,
+};
+
+export const EMPTY_STATE_STYLES = {
+    default: undefined as undefined | NestedCSSProperties,
+    hover: undefined as undefined | NestedCSSProperties,
+    focus: undefined as undefined | NestedCSSProperties,
+    clickFocus: undefined as undefined | NestedCSSProperties,
+    keyboardFocus: undefined as undefined | NestedCSSProperties,
+    active: undefined as undefined | NestedCSSProperties,
+    visited: undefined as undefined | NestedCSSProperties,
+    allStates: undefined as undefined | NestedCSSProperties,
+};
+
```

Comment:
Too many `undefined`. Is it best practice? Shouldn't it be some type?
That is a bit surprising to set `undefined` in TypeScript.